nomad/grapl-core.nomad

```diff
@@ -0,0 +1,428 @@
+                    sidecar_service {
+                        proxy {
+                            # Connect to the Zero leader
+                            upstreams {
+                                destination_name = "dgraph-zero-leader-grpc-private"
+                                local_bind_port = 5080
+                            }
+
+                            # Connect this Alpha to Zero followers
+                            dynamic "upstreams" {
+                                iterator = zero_follower
+                                for_each = [for zero_follower in local.dgraph_zeros: zero_follower]
+
+                                content {
+                                    destination_name = "dgraph-zero-follower-${zero_follower.value.id}-grpc-private"
+                                    local_bind_port = zero_follower.value.grpc_private_port
+                                }
+                            }
+
+                            # Connect this Alpha to Other Alphas (but not to itself, obviously)
+                            dynamic "upstreams" {
+                                iterator = alpha_peer
+                                for_each = [for alpha_peer in local.dgraph_alphas: alpha_peer if alpha_peer.id != alpha.value.id]
+
+                                content {
+                                    destination_name = "dgraph-alpha-${alpha_peer.value.id}-grpc-private"
+                                    local_bind_port = alpha_peer.value.grpc_private_port
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            service {
+                name = "dgraph-alpha-${alpha.value.id}-grpc-public"
+                port = "${alpha.value.grpc_public_port}"
+
+                connect {
+                    sidecar_service { }
+                }
+            }
+
+            service {
+                name = "dgraph-alpha-${alpha.value.id}-http"
+                port = "${alpha.value.http_port}"
+
+                connect {
+                    sidecar_service { }
+                }
+            }
+        }
+    }
+
+    group "grapl-graph-merger" {
+        count = var.graph_mergers
+
+        network {
+            mode = "bridge"
+        }
+
+        task "graph-merger" {
+            driver = "docker"
+
+            config {
+                image = "${var.container_registry}/graph-merger:${var.tag}"
+                args = ["/graph-merger"]
+            }
+
+            env {
+                REDIS_ENDPOINT = "${var.redis_endpoint}"
+                MG_ALPHAS = "${local.alpha_grpc_connect_str}"
+                GRAPL_SCHEMA_TABLE = "${var.schema_table_name}"
+                AWS_REGION = "${var.aws_region}"
+                # https://github.com/grapl-security/grapl/blob/18b229e824fae99fa2d600750dd3b17387611ef4/pulumi/grapl/__main__.py#L165
+                DEST_BUCKET_NAME = "subgraphs-merged-bucket"
```

Comment:
This isn't going to be the right bucket name. Currently, we still prepend the stack name, but soon, it will have a dynamically-generated suffix from Pulumi.

Reply:
> This isn't going to be the right bucket name.

This was a section that I was struggling with, honestly. I was trying to understand, for our multi-tenant future, what role the deployment name prefix would play (which I suspected it wouldn't exist anymore). If we're going to make a dynamically generated bucketname, we can always pass it in as a variable.

Reply:
Is every tennant going to get their own bucket?

If not, then the we'll just need to accept whatever the right bucket name is from Pulumi when this is deployed.

On the other hand, if everyone is getting their own bucket, we have some more planning to do 😬

Reply:
I actually think we're not going to be using buckets anymore when we move to kafka (if I understand kafka correctly). In this future world, we'll have topics that we're pushing messages through (where messages are a combination of the data and tenant id information).

In the meantime, it's probably fine to leave this as-is as we'll be deleting it (unless I misunderstood. cc: @jgrillo-grapl )

Reply:
My point is that, regardless of when Kafka lands, this is _not_ fine to leave as-is... there is nowhere in our infrastructure, local or otherwise, where we have a bucket named simply "subgraphs-merged-bucket", therefore this will not work with our existing code.

Reply:
I think we can discuss this on the call tomorrow so it's more clear.

Reply:
Fair enough.